libimage/copier.go

```diff
@@ -167,39 +167,45 @@ var storageAllowedPolicyScopes = signature.PolicyTransportScopes{
 	},
 }
 
-// getDockerAuthConfig extracts a docker auth config from the CopyOptions.  Returns
-// nil if no credentials are set.
-func (options *CopyOptions) getDockerAuthConfig() (*types.DockerAuthConfig, error) {
-	authConf := &types.DockerAuthConfig{IdentityToken: options.IdentityToken}
-
-	if options.Username != "" {
-		if options.Credentials != "" {
-			return nil, errors.New("username/password cannot be used with credentials")
-		}
-		authConf.Username = options.Username
-		authConf.Password = options.Password
-		return authConf, nil
-	}
+// getDockerAuthConfig extracts a docker auth config. Returns nil if
+// no credentials are set.
+func getDockerAuthConfig(name, passwd, creds, idToken string) (*types.DockerAuthConfig, error) {
+	numCredsSources := 0
 
-	if options.Credentials != "" {
-		split := strings.SplitN(options.Credentials, ":", 2)
+	if name != "" {
+		numCredsSources++
+	}
+	if creds != "" {
+		split := strings.SplitN(creds, ":", 2)
 		switch len(split) {
 		case 1:
-			authConf.Username = split[0]
+			name = split[0]
 		default:
-			authConf.Username = split[0]
-			authConf.Password = split[1]
+			name = split[0]
+			passwd = split[1]
 		}
-		return authConf, nil
+		numCredsSources++
+	}
+	if idToken != "" {
+		numCredsSources++
+	}
+	authConf := &types.DockerAuthConfig{
+		Username:      name,
+		Password:      passwd,
+		IdentityToken: idToken,
 	}
 
-	// We should return nil unless a token was set.  That's especially
-	// useful for Podman's remote API.
-	if options.IdentityToken != "" {
-		return authConf, nil
+	// Cannot use the multiple credential sources.
+	if numCredsSources > 1 {
+		return nil, errors.New("Cannot use the multiple credential sources")
```

Comment:
```suggestion
		return nil, errors.New("cannot use the multiple credential sources")
```
is the Go convention I think